src/shared/udev-util.c

```diff
@@ -108,8 +110,31 @@ int udev_parse_config_full(
         return 0;
 }
 
+static int device_new_from_dev_path(const char *devlink, sd_device **ret_device) {
+        struct stat st;
+        int r;
+
+        assert(devlink);
+
+        r = stat(devlink, &st);
+        if (r < 0 && errno == ENOENT)
+                return -ENOENT;
```

Comment:
so this function logs on all errors, but one: ENOENT. This deserves a comment, somewhere at the top of the function since this is not how we usually do stuff. usually our functions either log about all errors or about none.

(given this is an internal function it's definitely fine to deviate from our usual rule, but it should have a comment)

Reply:
using a single return log_full_errno as suggested below, so now all return paths log